aimmo-game/simulation/world_map.py

```diff
@@ -44,14 +43,23 @@ def generate_empty_map(cls, height, width, settings):
 
         (min_x, max_x, min_y, max_y) = WorldMap._min_max_from_dimensions(height, width)
         grid = {}
-        for x in xrange(min_x, max_x + 1):
-            for y in xrange(min_y, max_y + 1):
+        for x in range(min_x, max_x + 1):
+            for y in range(min_y, max_y + 1):
                 location = Location(x, y)
                 grid[location] = Cell(location)
         return cls(grid, new_settings)
 
     def all_cells(self):
-        return self.grid.itervalues()
+        '''
+        Written to support both python3 and python2. itervalues() became obsolete
+        in py3 and values() returns dictionary views, which are iterators.
+        :return: Iterator (dictionary or iterator) of all cells in the grid.
+        '''
+        try:
+            values = self.grid.itervalues()
+        except:
```

Comment:
you could try to except for a specific error type, less messy - in case the problem is not linked to the version of python for instance?